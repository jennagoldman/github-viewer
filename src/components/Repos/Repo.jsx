import React from 'react';

import PropTypes from 'prop-types';

const Repo = ({ name, html_url }) => (
  <>
    <a href={html_url} target="_blank" rel="noopener noreferrer">{name}</a>
  </>
);

Repo.propTypes = {
  html_url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

export default Repo;
