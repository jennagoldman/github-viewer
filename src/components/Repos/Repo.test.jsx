import React from 'react';
import { shallow } from 'enzyme';
import Repo from './Repo.jsx';

describe('Repo', () => {
  it('matches a snapshot', () => {

    const wrapper = shallow(<Repo name="rock-paper-scissors" html_url="https://github.com/jennagoldman/rock-paper-scissors" />);
    expect(wrapper).toMatchSnapshot();
  });
});
