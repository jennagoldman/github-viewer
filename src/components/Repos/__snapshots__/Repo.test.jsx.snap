// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo matches a snapshot 1`] = `
<Fragment>
  <a
    href="https://github.com/jennagoldman/rock-paper-scissors"
    rel="noopener noreferrer"
    target="_blank"
  >
    rock-paper-scissors
  </a>
</Fragment>
`;
